dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java

```diff
@@ -89,6 +98,19 @@ public Response personalizePageContainers (@Context final HttpServletRequest  re
             throw new BadRequestException("Does not exists a Persona with the tag: " + personalizationPersonaPageForm.getPersonaTag());
         }
 
+        final String pageId = personalizationPersonaPageForm.getPageId();
+
+        if (!UtilMethods.isSet(pageId)) {
+            throw new BadRequestException(
+                    "Page parameter is missing");
+        }
+
+        final Contentlet pageContentlet = contentletAPI.findContentletByIdentifierAnyLanguage(pageId);
+        if(!permissionAPI.doesUserHavePermission(pageContentlet, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndRoles)){
+            Logger.warn(PersonalizationResource.class,String.format("User `%s` does not have edit permission over page `%s` therefore personalization isn't allowed.  ",user.getUserId(), pageId));
+            return Response.status(Status.UNAUTHORIZED).build();
```

Comment:
When no permissions we use `Status.FORBIDDEN`

Reply:
done!